parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -356,6 +357,8 @@ pub mod pallet {
 			meter: &mut frame_support::weights::WeightMeter,
 			_: &mut [u8; 32],
 		) -> Result<bool, ProcessMessageError> {
+			// Make sure the bridge not halted
+			Self::ensure_not_halted().map_err(|_| ProcessMessageError::Yield)?;
```

Comment:
I looked the `Yield` error description:

> The queue wants to give up its current processing slot. Hints the message processor to cease servicing this queue and proceed to the next one. This is seen as a hint, not an instruction. Implementations must therefore handle the case that a queue is re-serviced within the same block after yielding. A queue is not required to yield again when it is being re-serviced withing the same block.

Does the implementation of this message queue discard this message?

Reply:
No, it's not discarded and will be reattempted later. There are some other errors like `BadFormat` or `Corrupt`
https://github.com/paritytech/polkadot-sdk/blob/b728724318d65bf8d61831aac35c286727b41b60/substrate/frame/support/src/traits/messages.rs#L32-L36
will be discarded.